lib/Runtime/Types/ES5ArrayTypeHandler.cpp

```diff
@@ -271,25 +279,34 @@ namespace Js
             doneConversion = true;
 #endif
         }
+#ifdef _MSC_VER
         __finally
         {
             if (AbnormalTermination())
```

Comment:
IMO, TryFinally wouldn't fit here as is. `if (AbnormalTermination())` 

If SEH is not needed `try/catch(...)` should be fine. See https://github.com/Microsoft/ChakraCore/pull/726/files#diff-ffccee5f5c8edef39aeaad0f043b66d8R252